client/client.go

```diff
@@ -57,52 +59,65 @@ func (cli *Client) call(ctx context.Context, paramsRawVals []interface{}, interf
 	if err != nil {
 		return nil, err
 	}
+
+	refOption := cli.refOpts[common.ServiceKey(interfaceName, group, version)]
+	if refOption == nil {
+		return nil, fmt.Errorf("no service found for %s/%s:%s, please check if the service has been registered", group, interfaceName, version)
+	}
 	// todo: move timeout into context or invocation
-	return cli.invoker.Invoke(ctx, inv), nil
+	return refOption.invoker.Invoke(ctx, inv), nil
 
 }
 
-func (cli *Client) CallUnary(ctx context.Context, req, resp interface{}, interfaceName, methodName string, opts ...CallOption) error {
-	res, err := cli.call(ctx, []interface{}{req, resp}, interfaceName, methodName, constant.CallUnary, opts...)
+func (cli *Client) CallUnary(ctx context.Context, req, resp interface{}, interfaceName, methodName string, group string, version string, opts ...CallOption) error {
+	res, err := cli.call(ctx, []interface{}{req, resp}, interfaceName, methodName, group, version, constant.CallUnary, opts...)
 	if err != nil {
 		return err
 	}
 	return res.Error()
 }
 
-func (cli *Client) CallClientStream(ctx context.Context, interfaceName, methodName string, opts ...CallOption) (interface{}, error) {
-	res, err := cli.call(ctx, nil, interfaceName, methodName, constant.CallClientStream, opts...)
+func (cli *Client) CallClientStream(ctx context.Context, interfaceName, methodName, group, version string, opts ...CallOption) (interface{}, error) {
+	res, err := cli.call(ctx, nil, interfaceName, methodName, group, version, constant.CallClientStream, opts...)
 	if err != nil {
 		return nil, err
 	}
 	return res.Result(), res.Error()
 }
 
-func (cli *Client) CallServerStream(ctx context.Context, req interface{}, interfaceName, methodName string, opts ...CallOption) (interface{}, error) {
-	res, err := cli.call(ctx, []interface{}{req}, interfaceName, methodName, constant.CallServerStream, opts...)
+func (cli *Client) CallServerStream(ctx context.Context, req interface{}, interfaceName, methodName, group, version string, opts ...CallOption) (interface{}, error) {
+	res, err := cli.call(ctx, []interface{}{req}, interfaceName, methodName, group, version, constant.CallServerStream, opts...)
 	if err != nil {
 		return nil, err
 	}
 	return res.Result(), res.Error()
 }
 
-func (cli *Client) CallBidiStream(ctx context.Context, interfaceName, methodName string, opts ...CallOption) (interface{}, error) {
-	res, err := cli.call(ctx, nil, interfaceName, methodName, constant.CallBidiStream, opts...)
+func (cli *Client) CallBidiStream(ctx context.Context, interfaceName, methodName, group, version string, opts ...CallOption) (interface{}, error) {
+	res, err := cli.call(ctx, nil, interfaceName, methodName, group, version, constant.CallBidiStream, opts...)
 	if err != nil {
 		return nil, err
 	}
 	return res.Result(), res.Error()
 }
 
-func (cli *Client) Init(info *ClientInfo) error {
+func (cli *Client) Init(interfaceName string, info *ClientInfo, opts ...ReferenceOption) (string, string, error) {
```

Comment:
```ClientInfo``` contains ```InterfaceName```. And we also need to consider about non-idl API style since this function is also important.

Reply:
Got